src/engraving/dom/textbase.cpp

```diff
@@ -3315,17 +3315,23 @@ void TextBase::drawEditMode(Painter* p, EditData& ed, double currentViewScaling)
     pen.setJoinStyle(PenJoinStyle::MiterJoin);
     p->setPen(pen);
 
-    // Don't draw cursor if there is a selection
-    if (!cursor->hasSelection()) {
+    // Don't draw cursor if it's dynamic or if there is a selection
+    if (!ed.element->isDynamic() && !cursor->hasSelection()) {
```

Comment:
This is questionable; for existing scores, dynamics may also contain normal text besides the dynamic symbol. So the user should also be able to edit that normal text, so a text cursor would be required.